src/main/java/java_echo_server/App.java

```diff
@@ -1,14 +1,57 @@
-/*
- * This Java source file was generated by the Gradle 'init' task.
- */
 package java_echo_server;
 
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.PrintWriter;
+import java.net.ServerSocket;
+import java.net.Socket;
+
 public class App {
-    public String getGreeting() {
-        return "Hello world.";
-    }
+    private static ServerSocket serverSocket;
+    private static Socket clientSocket;
 
     public static void main(String[] args) {
-        System.out.println(new App().getGreeting());
+        int portNumber = setPortNumber(args);
+        Client client = createClientSocketWrapper(portNumber);
+
+        EchoServer echoServer = new EchoServer(client);
+
+        do {
```

Comment:
Why are you using a `do while` loop here? Does `echoServer.start()` need to run at least once, even when the `while` condition returns false?